cg/models/nf_analysis.py

```diff
@@ -24,11 +25,19 @@ class NextflowSample(BaseModel):
     fastq_forward: List[str]
     fastq_reverse: List[str]
 
+    @validator("fastq_reverse", "fastq_forward")
+    def fastq_not_empty(cls, fastq) -> List[str]:
+        """Verify that fastq files are provided."""
+        if len(fastq) == 0:
+            raise SampleSheetError("Fastq files not provided")
+        return fastq
+
     @validator("fastq_reverse")
-    def fastq_forward_reverse_length_match(cls, fastq_reverse: List[str], values: dict) -> str:
+    def fastq_forward_reverse_length_match(cls, fastq_reverse, values: dict) -> List[str]:
         """Verify that the number of fastq files is the same for R1 and R2."""
-        assert len(fastq_reverse) == len(values.get("fastq_forward")) or len(fastq_reverse) == 0
-        return "Fastq file length for forward and reverse do not match"
+        if len(fastq_reverse) != len(values.get("fastq_forward")):
+            raise SampleSheetError("Fastq file length for forward and reverse do not match")
+        return fastq_reverse
```

Comment:
Do they need to return anything?

Reply:
I think so? at least when giving specific fields. at least that's how I've always seen it